src/Marten/Services/UnitOfWork.cs

```diff
@@ -342,7 +336,73 @@ public void Eject<T>(T document)
             {
                 operations.Remove(operation);
             }
-           
+        }
+
+        private class StorageOperationComparer : IComparer<IStorageOperation>
+        {
+            private readonly Type[] _topologicallyOrderedTypes;
+
+            public StorageOperationComparer(Type[] topologicallyOrderedTypes)
+            {
+                _topologicallyOrderedTypes = topologicallyOrderedTypes;
+            }
+
+            public int Compare(IStorageOperation x, IStorageOperation y)
+            {
+                if (x == y)
+                {
+                    return 0;
+                }
+
+                if (x?.DocumentType == null || y?.DocumentType == null)
+                {
+                    return 0;
```

Comment:
Are you sure that if one of the storage operations are `null` then we should return `0`?

Reply:
To be honest, I was not sure what to do here. I choose 0 to try and maintain relative order of operations in this case, as without knowing what they actually are the order it _should_ be feels undefined